sdk/program-entrypoint/src/lib.rs

```diff
@@ -279,6 +279,85 @@ macro_rules! custom_panic_default {
     };
 }
 
+/// This is a more space efficient implementation of custom panic.
+/// It is about 1kb in size and depends on the unstable feature `panic_info_message`,
+/// which is going to be stabilized in Rust 1.84.
+///
+/// It works just like the default custom panic, except that dynamic generated error
+/// messages are not shown. For instance, trying to access an index out of bounds of a vector
+/// would give us `index Y is out of bounds for length X`. As the length is only know at runtime
+/// time, this message is elided.
+///
+/// All messages known at compile time are correctly displayed, e.g. `called unwrap in a None`,
+/// file names, line and column numbers.
+#[macro_export]
+macro_rules! custom_panic_space_efficient {
+    () => {
+        #[cfg(all(not(feature = "custom-panic"), target_os = "solana"))]
```

Comment:
I think we need to remove the "gate" on the `not(feature = "custom-panic")`. Since the `entrypoint!` macro always uses the `custom_panic_default!`, the only way to "disable" the default panic handler is to enable the feature `"custom-panic"` on the program. This would also disable the `custom_panic_space_efficient!` as it is.

Removing the `not(feature = "custom-panic")` in this case would solve this – a program can enable the `"custom-panic"` feature to disable the default and then use `custom_panic_space_efficient!`.

Reply:
My idea is that the space efficient implementation will replace the existing one, once we have Rust 1.84 or higher in the platform tools.

What do you think of this?